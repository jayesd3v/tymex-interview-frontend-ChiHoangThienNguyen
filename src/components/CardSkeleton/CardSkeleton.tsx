import { Col, Placeholder } from 'react-bootstrap';

const CardSkeleton = () => {
    return (
        <Col xs={12} md={6} lg={4} xl={3} data-testid='skeleton-card'>
            <div className='character-card p-2 rounded mb-3'>
                <Placeholder as='div' animation='glow' className='w-100'>
                    <Placeholder
                        xs={12}
                        style={{
                            height: '150px',
                            width: '100%',
                        }}
                    />
                </Placeholder>
                <Placeholder as='div' animation='glow' className='w-100 mt-4'>
                    <Placeholder
                        xs={12}
                        style={{
                            height: '20px',
                            width: '100%',
                        }}
                    />
                </Placeholder>

                <Placeholder as='div' animation='glow' className='w-100 mt-4'>
                    <Placeholder
                        xs={12}
                        style={{
                            height: '20px',
                            width: '100%',
                        }}
                    />
                </Placeholder>
            </div>
        </Col>
    );
};

export default CardSkeleton;
