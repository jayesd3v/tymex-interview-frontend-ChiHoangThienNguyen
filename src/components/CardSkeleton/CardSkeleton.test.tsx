import { render } from "@testing-library/react";
import CardSkeleton from "./CardSkeleton";

describe('CardSkeleton tests', () => {
    test('snapshot test', () => {
        const { container } = render(<CardSkeleton />);
        expect(container).toMatchSnapshot();
    });
});
