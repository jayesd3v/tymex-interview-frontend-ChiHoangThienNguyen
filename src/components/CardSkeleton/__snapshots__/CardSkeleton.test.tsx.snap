// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkeleton tests snapshot test 1`] = `
<div>
  <div
    class="col-xl-3 col-lg-4 col-md-6 col-12"
    data-testid="skeleton-card"
  >
    <div
      class="character-card p-2 rounded mb-3"
    >
      <div
        class="w-100 placeholder-glow"
      >
        <span
          class="col-12 placeholder"
          style="height: 150px; width: 100%;"
        />
      </div>
      <div
        class="w-100 mt-4 placeholder-glow"
      >
        <span
          class="col-12 placeholder"
          style="height: 20px; width: 100%;"
        />
      </div>
      <div
        class="w-100 mt-4 placeholder-glow"
      >
        <span
          class="col-12 placeholder"
          style="height: 20px; width: 100%;"
        />
      </div>
    </div>
  </div>
</div>
`;
